cpp/tests/CMakeLists.txt

```diff
@@ -44,6 +44,7 @@ find_package(Threads)
 target_link_libraries(tests PRIVATE Threads::Threads ${CMAKE_PROJECT_NAME} ${JSONCPP_TARGET} ${GOOGLETEST_TARGET})
 open3d_show_and_abort_on_warning(tests)
 open3d_set_global_properties(tests)
+open3d_link_3rdparty_libraries(tests)
```

Comment:
I would remove this and change the tests. Typically, we only link `tests` to Open3D but not 3rd-party libraries, such that the `tests` works like a user's program linking to Open3D.

Reply:
Got it, without the `#pragmas` we don't even need to pass the `-fopenmp` flag to the compiler, so we don't need this anymore